doc/development/style_guide.md

```diff
@@ -0,0 +1,870 @@
+  in the manual
+- the **second** keyword is the overall topic which goes to the
+  [topic index](https://grass.osgeo.org/grass-stable/manuals/topics.html) in the
+  manual
+- the **third** (and more) keyword goes to the
+  [keyword index](https://grass.osgeo.org/grass-stable/manuals/keywords.html) in
+  the manual
+
+These index manual pages are autogenerated during the build process of GRASS
+GIS.
+
+#### Lazy import of optional dependencies
+
+For optional dependencies, import only after the _gs.parser_ call. In that way the
+tool can be safely compiled even if the dependency is not installed.
+
+```python
+def main():
+    options, flags = gs.parser()
+    try:
+        import pandas as pd  # noqa: E402
+    except ModuleNotFoundError:
+      gs.fatal(_("Pandas library not installed"))
+```
+
+#### Tool name
+
+Try to use names which describe shortly the intended purpose of the tool.
+
+The first letters for the tool name should be:
+
+```text
+d.    - display commands
+db.   - database commands
+g.    - general GIS management commands
+i.    - imagery commands
+m.    - miscellaneous tool commands
+ps.   - postscript commands
+r.    - raster commands
+r3.   - raster3D commands
+v.    - vector commands
+t.    - temporal commands
+```
+
+Some additional naming conventions
+
+- export modules: (type).out.(format) eg: _r.out.arc_, _v.out.ascii_
+- import module: (type).in.(format) eg: _r.in.arc_, _v.in.ascii_
+- conversion modules: (type).to.(type) eg: _r.to.vect_, _v.to.rast_,
+  _r3.to.rast_
+
+Avoid module names with more than two dots in the name. Example: instead of
+_r.to.rast3.elev_ use _r.to.rast3elev_.
+
+#### Data processing history
+
+Tools should record processing history to the output data. For vectors:
+
+```python
+gs.vector_history(output)
+```
+
+For rasters:
+
+```python
+gs.raster_history(output, overwrite=True)
+```
+
+### Developing GRASS GUI
+
+Follow
+[wxPython style guide](https://wiki.wxpython.org/wxPython%20Style%20Guide).
+
+Please use the following docstring template:
+
+```py
+"""!
+@package dir.example
+
+@brief Short example package description
+
+Classes:
+ - example::ExampleClass
+
+(C) 2024 by the GRASS Development Team
+
+This program is free software under the GNU General Public License
+(>=v2). Read the file COPYING that comes with GRASS for details.
+
+@author First Author <first somewhere.com>
+@author Second Author <second somewhere.com>
+@author Some Other <third somewhere.com> (some particular change)
+"""
+```
+
+#### Translations
+
+To enable translating of messages to other languages, use full strings, e.g.
+(good example):
+
+```python
+if ...:
+    win.SetLabel(_("Name for new 3D raster map to create"))
+else:
+    win.SetLabel(_("Name for new raster map to create"))
+```
+
+instead of constructing string from several parts (bad example):
+
+```python
+if ...:
+    maplabel = 'raster map'
+else:
+    maplabel = '3D raster map'
+win.SetLabel(_("Name for new {} to create").format(maplabel))
+```
```

Comment:
When reading great recent docs in other softwares, I often see that the bad examples have comments inside the code block, or even inline, mentionnons it is bad and to not use, in addition to having it said in the docs. Probably because people don't always read and copy code blocks, or for scraping/AI purposes. Sometimes with a good formatting the lines are in red background, but I don't think simple markdown code blocks allow it.